### PR TITLE
Adds suggested naming convention for container interop implementations

### PR DESCRIPTION
**Pull request by carnage**
not sure how the other commits got dragged along :s

### CODE REVIEW

**Comment by michaelcullum** [comment]
@mnapoli

File: proposed/container.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ Users of dependency injections containers (DIC) are referred to as `user`.
 
 - `has` takes one unique parameter: an entry identifier. It MUST return `true`
   if an entry identifier is known to the container and `false` if it is not.
+  `has($id)` returning true does not mean that `get($id)` will not throw an exception.
+  It does however mean that `get($id)` will not throw a `NotFoundException`.
```
Comment (mnapoli) [inline proposed/container.md]: :+1: for this part as this is backporting this PR to the standard: https://github.com/container-interop/container-interop/pull/45

File: proposed/container.md [hunk 2]
```diff
@@ -81,6 +83,10 @@ into the same container (or another container) instead of the delegate container
 The interfaces and classes described as well as relevant exception are provided as part of the
 [psr/container](https://packagist.org/packages/psr/container) package. (still to-be-created)
 
+Packages providing a psr container implementation should declare that they provide psr/container-implementation 1.0.0
+
+Projects requiring an implementation should require psr/container-implementation 1.0.0
+
```
Comment (mnapoli) [inline proposed/container.md]: First time I see that `psr/log` has a `psr/log-implementation`, and I guess it's the same for PSR-7. OK for this if it's common for PSRs.

**Comment by mnapoli** [comment]
Ah well I just realized you mentionned other commits being included in this PR. All in all this PR is fine by me, ping @moufmouf.

**Comment by michaelcullum** [comment]
Bump @mnapoli @moufmouf @php-fig/psr-11. What do you want to do with this?

**Comment by mnapoli** [comment]
LGTM, it's worth merging it IMO.

**Pull request merged**